packages/store/addon/-private/system/promise-proxies.js

```diff
@@ -88,3 +89,34 @@ export function promiseArray(promise, label) {
     promise: Promise.resolve(promise, label),
   });
 }
+
+// constructor is accessed in some internals but not including it in the copyright for the deprecation
+const ALLOWABLE_METHODS = ['constructor', 'then', 'catch', 'finally'];
+
+export function deprecatedPromiseObject(promise) {
+  const handler = {
+    get(target, prop) {
+      if (!ALLOWABLE_METHODS.includes(prop)) {
+        deprecate(
+          `Accessing ${prop} is deprecated.  Only available methods to access on a promise returned from model.save() are .then, .catch and .finally`,
+          false,
+          {
+            id: 'ember-data:model-save-promise',
+            until: '5.0',
+            for: '@ember-data/store',
+            since: {
+              available: '4.4',
+              enabled: '4.4',
+            },
+          }
+        );
+      }
+
+      /* global Reflect */
+      return Reflect.get(...arguments).bind(target);
```

Comment:
when used with ember-concurrency, trying to access a `__ec_cancel__` property on the promise object triggers the following error:

```
promise-proxies.js:128 Uncaught TypeError: Cannot read properties of undefined (reading 'bind')
    at Object.get (promise-proxies.js:128:39)
    at TaskInstanceExecutor.handleYieldedValue (executor.js:233:34)
    at TaskInstanceExecutor.handleResolvedContinueValue (executor.js:160:10)
    at TaskInstanceExecutor.proceedSync (executor.js:112:12)
    at TaskInstanceExecutor.start (executor.js:52:10)
    at TaskInstance.start (base.js:77:19)
    at eval (refresh.js:24:65)
    at Array.forEach (<anonymous>)
    at Refresh.process (refresh.js:24:28)
    at EmberScheduler.refresh (scheduler.js:51:34)
```

this appears to be caused by `Reflect.get(...arguments)` returning `undefined` here.

Reply:
unfortunately it seems like setting `compatWith: '4.4'` does not resolve this issue. this can be reproduced by the frontend test suite in https://github.com/rust-lang/crates.io/pull/4801

Reply:
@Turbo87 did you ever find a way around this issue?

Reply:
I've opened https://github.com/emberjs/data/pull/8000 to fix it, but it's still not merged :-/